test/Test/Pos/Block/LogicSpec.hs

```diff
@@ -0,0 +1,133 @@
+import           Test.Hspec.QuickCheck      (prop)
+import           Test.QuickCheck            (Gen, Property, Testable, arbitrary, choose,
+                                             counterexample, elements, forAll, generate,
+                                             listOf, listOf1, oneof, property)
+
+import           Pos.Binary.Class           (biSize)
+import qualified Pos.Binary.Class           as Bi
+import           Pos.Block.Arbitrary        ()
+import           Pos.Block.Logic            (createMainBlockPure)
+import qualified Pos.Communication          ()
+import           Pos.Constants              (blkSecurityParam, genesisMaxBlockSize)
+import           Pos.Core                   (SlotId (..))
+import           Pos.Crypto                 (SecretKey)
+import           Pos.Ssc.Class              (Ssc (..), sscDefaultPayload)
+import           Pos.Ssc.GodTossing         (GtPayload (..), SscGodTossing,
+                                             commitmentMapEpochGen, mkVssCertificatesMap,
+                                             vssCertificateEpochGen)
+import           Pos.Txp.Core               (TxAux)
+import           Pos.Types                  (BlockHeader, MainBlock, ProxySKEither,
+                                             ProxySKHeavy, SmallGoodTx (..),
+                                             goodTxToTxAux)
+import           Pos.Update.Core            (UpdatePayload (..))
+import           Pos.Util.Arbitrary         (makeSmall)
+
+
+spec :: Spec
+spec = describe "Block.Logic" $ do
+
+    -- Sampling the minimum empty block size
+    (sk0,prevHeader0) <- runIO $ generate arbitrary
+
+    -- We multiply by 1.5 because there are different possible
+    -- combinations of empty constructors and there's no out-of-box
+    -- way to get maximum of them. Some settings produce 390b empty
+    -- block, some -- 431b.
+    let emptyBSize0 :: Byte
+        emptyBSize0 = biSize (noSscBlock infLimit [] prevHeader0 sk0 def []) -- in bytes
+        emptyBSize :: Integral n => n
+        emptyBSize = round $ (1.5 * fromIntegral emptyBSize0 :: Double)
+
+    describe "createMainBlockPure" $ do
+        prop "empty block size is sane" $ emptyBlk $ \blk ->
+            let s = biSize blk
+            in counterexample ("Real block size: " <> show s) $
+               s <= 500 && s <= genesisMaxBlockSize
+        prop "doesn't create blocks bigger than the limit" $
+            forAll (choose (emptyBSize, emptyBSize * 10)) $ \(fromBytes -> limit) ->
+            forAll arbitrary $ \(prevHeader, sk, updatePayload, proxyCerts) ->
+            forAll validGtPayloadGen $ \(gtPayload, slotId) ->
+            forAll (makeSmall $ listOf1 genTx) $ \txs ->
+            let blk = producePureBlock limit prevHeader txs Nothing slotId
+                                       proxyCerts gtPayload updatePayload sk
+            in leftToCounter blk $ \b ->
+                let s = length (Bi.encode b)
```

Comment:
Here you can also use `biSize` and not use `fromIntegral` and print it using `memory` formatter (or `unitBuilder`) which is smart.